packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -102,16 +104,8 @@ class Checkbox extends StatefulWidget {
   /// or macOS, following Material design's
   /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
   ///
-  /// On iOS and macOS, this constructor creates a [CupertinoCheckbox], which has
-  /// matching functionality and presentation as Material checkboxes, and are the
-  /// graphics expected on iOS. On other platforms, this creates a Material
-  /// design [Checkbox].
-  ///
-  /// If a [CupertinoCheckbox] is created, the following parameters are ignored:
-  /// [mouseCursor], [fillColor], [hoverColor], [overlayColor], [splashRadius],
-  /// [materialTapTargetSize], [visualDensity], [isError]. However, [shape] and
-  /// [side] will still affect the [CupertinoCheckbox] and should be handled if
-  /// native fidelity is important.
+  /// Creates a checkbox that looks and feels native when the [ThemeData.platform]
+  /// is iOS or macOS, otherwise a Material Design checkbox is created.
```

Comment:
I think we use the same kind of wording for Switch, but it sounds confusing to me, like it's only "native" on iOS and macOS. Maybe instead:

Creates a checkbox that adapts to the look and feel of [ThemeData.platform] for iOS and macOS, and otherwise builds a Material Design checkbox.